scripts/fig1b-modality_barchart.R

```diff
@@ -0,0 +1,138 @@
+  # make sure all bulk and spatial get designated with cell or nucleus
+  # also rename to be more specific when creating legend
+  dplyr::mutate(
+    seq_unit = dplyr::case_when(
+      scpca_sample_id %in% all_single_cell ~ "Single-cell",
+      scpca_sample_id %in% all_single_nuc ~ "Single-nuclei"
+    )
+  ) |> 
+  # first combine all modalities for each sample id into one list
+  # make sure to keep seq unit
+  dplyr::group_by(scpca_sample_id, seq_unit) |>
+  dplyr::summarise(modality = paste(modality, collapse = ";")) |> 
+  # split any lists of modalities so we have one row per sample per modality
+  dplyr::mutate(modality = stringr::str_split(modality, ";")) |> 
+  tidyr::unnest(modality) |>
+  dplyr::group_by(modality) |> 
+  # get the total for each modality to use for specifying order in plot
+  dplyr::add_tally(name = "total_per_modality") |>
+  # add a column to help pull out additional modalities into its own facet 
+  dplyr::mutate(additional_mods = ifelse(modality == "Single suspension", "All samples", "Samples with additional modalities"),
+                modality = as.factor(modality),
+                # set custom order
+                # with cite and with cell hash should be together 
+                modality = forcats::fct_relevel(modality,
+                                                "Single suspension",
+                                                "Bulk",
+                                                "Spatial transcriptomics",
+                                                "With CITE-seq",
+                                                "With cell hashing"
+                ))
+
+# Plot -------------------------------------------------------------------------
+
+# get the colors for each suspension type
+suspension_colors <- suspension_palette$color |> 
+  purrr::set_names(suspension_palette$suspension_type)
+
+# bar chart showing the total number of samples for each modality
+# the bars are colored by what suspension type that modality is composed of
+ggplot(filtered_modality_df, aes(x = modality, fill = seq_unit)) +
+  geom_bar(stat = "count", color = "black") +
+  # split the graph to designate all samples vs. those with additional modalities
+  facet_grid(cols = vars(additional_mods),
+             scales = "free",
+             space = "free") +
+  theme_classic() + 
+  # add label to middle of each bar 
+  geom_text(aes(label = after_stat(count)), stat = "count", position = position_stack(vjust = 0.5)) +
+  labs(
+    x = "",
+    y = "Number of samples",
+    fill = "Suspension type"
+  ) + 
+  scale_fill_manual(values= suspension_colors) +
+  theme(legend.position = c(.8,.9),
+        legend.direction = "horizontal",
+        legend.box.background = element_rect(colour = "black"),
+        text = element_text(size = 14),
+        axis.text.x = element_text(angle = 45, hjust = 1)) +
+  guides(fill = guide_legend(title.position = "top", title.hjust = 0.5))
+
+ggsave(output_plot_file)
```

Comment:
Can we specify a width/height here to ensure reproducibility? The default is 7x7, so maybe just pop those in since it looks fine. IMO this is a good practice to ensure that the chosen text sizes will definitely work!